internal/controller/storageclient_controller.go

```diff
@@ -697,6 +702,43 @@ func removeStorageClaimAsOwner(obj client.Object) {
 	}
 }
 
+func (r *storageClientReconcile) reconcileResourcesByGVK(
+	resources [][]byte,
+	obj client.Object,
+	combinedErr error,
+) {
+	retainKeys := make(map[types.NamespacedName]bool, len(resources))
+	gvk := obj.GetObjectKind().GroupVersionKind()
+	for idx := range resources {
+		// object supplied to reconcile mutates it, we either need to send
+		// fresh copy or zero out fields on the mutated object, doing the former
+		// as we don't know the fields that needs to be zeroed for every concrete object
+		var newObj client.Object = reflect.New(reflect.TypeOf(obj).Elem()).Interface().(client.Object)
```

Comment:
Please separate the steps into their own lines. This kind of chaining is unreadable

Reply:
```
objType := reflect.TypeOf(obj).Elem()
copyAsAny := reflect.New(type),Interface()
copy := untypedCopy.(client.Object)
```